multy_core/src/bitcoin/bitcoin_transaction.cpp

```diff
@@ -668,20 +668,23 @@ void BitcoinTransaction::update_state()
     {
         sign();
         BitcoinBytesCountStream counter_stream;
-        serialize_to_stream(&counter_stream);
+        serialize_to_stream(&counter_stream, true);
         const uint64_t tx_size = counter_stream.get_bytes_count();
 
         const BigInt tx_cost = tx_size * *m_fee->amount_per_byte;
         const BigInt current_fee = calculate_diff();
         const BigInt remainder = current_fee - tx_cost;
-        if (remainder > 0)
+        //NOTE: with SegWit transaction limit 100 satoshi
+        if (remainder > 150 && m_is_segwit_transaction == 0)
```

Comment:
0. Don't use m_is_segwit_transaction, use `is_segwit()`.
1. Simplify and throw away any repeating parts:
```
const int64_t average_input_size = is_segwit() ? 100 : 150;
const int64_t min_tx_cost_per_byte = 1;
if (remainder > average_input_size * min_tx_cost_per_byte)
{
    // do stuff
}
```
2. Put a long descriptive comment on why we do that.